### ISSUE TITLE
Change the yml file structure

### ISSUE THREAD

**Issue by HuaDeity** [catppuccin/lazygit#7]
It will be great if we can change the themes/*.yml structure from
``` yml
theme:
  lightTheme: false 
  activeBorderColor:
    - "#a6e3a1" # Green
    - bold
```
to
```yml
gui:
  theme:
    lightTheme: false 
    activeBorderColor:
      - "#a6e3a1" # Green
      - bold
```

And then we can just use [overriding-default-config-file-location](https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md#overriding-default-config-file-location)
```bash
lazygit --use-config-file="$HOME/.config/lazygit/config.yml,themes/*.yml"
or
LG_CONFIG_FILE="$HOME/.config/lazygit/config.yml,themes/*.yml" lazygit
```
to merge the theme config with base config.

**Comment by nullishamy** [comment]
Is this a breaking change? Can you reference some changelogs from lazygit which mentions this config change so we can investigate it?

**Comment by HuaDeity** [comment]
No, this is not a major change. I just discovered this method during the use.

Because the copy and paste method is relatively traditional, using files directly has some advantages. It can facilitate timely access to theme updates through version control tools, and it can also automatically use corresponding Mocha/Latte themes by obtaining the system's dark/light themes.

The current file structure lacks

```yaml

gui:

```

which causes lazygit to be unable to recognize the merged configuration file.

**Comment by nullishamy** [comment]
This was intentional, because the default config always has the `gui` part set. This way, users can override only the theme block, which may be below other config options under `gui`. I'm not sure how adding this to the repo would be an improvement, can you clarify?

**Comment by HuaDeity** [comment]
In my test, if there is a basic config file, such as `$HOME/.config/lazygit/config.yml`, which contains some GUI configurations:
```yml
gui:
  windowSize: 'full'
  language: 'en'
```

Then downloaded the theme file `mocha.yml` and used the command:
```bash
lazygit --use-config-file="$HOME/.config/lazygit/config.yml,mocha.yml"
or
LG_CONFIG_FILE="$HOME/.config/lazygit/config.yml,mocha.yml" lazygit
```

Even if `mocha.yml` contains the `gui` field, the GUI configuration of both files will be merged without overwriting each other, and other configurations will also take effect.

We can write two configuration methods in the `README`:

1. Download the theme file and use the command above.
2. Add a 'theme' field under 'gui' in the original configuration file.

This way, users can choose a configuration method according to their usage habits.

**Comment by nullishamy** [comment]
Won't specifying the config file to use via args / env stop the application of the normal file? Or does it always evaluate `~/.config/lazygit/config.yml` no matter what?

**Comment by nullishamy** [comment]
Got some clarification from the Discord, and I believe I understand your concern. Should be fixed in #10!

**Comment by HuaDeity** [comment]
Thank you.
This is an awesome solution.
